<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>gui</class>
 <widget class="QWidget" name="gui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>494</width>
    <height>541</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Magneto</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>Editor</string>
  </property>
  <widget class="QListWidget" name="faceList">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>60</y>
     <width>141</width>
     <height>331</height>
    </rect>
   </property>
  </widget>
  <widget class="QListWidget" name="halfEdgeList">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>60</y>
     <width>151</width>
     <height>331</height>
    </rect>
   </property>
  </widget>
  <widget class="QListWidget" name="vertexList">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>60</y>
     <width>141</width>
     <height>331</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="faceLabel">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>10</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Face</string>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
  </widget>
  <widget class="QLabel" name="halfEdgeLabel">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>10</y>
     <width>141</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="text">
    <string>HalfEdge</string>
   </property>
  </widget>
  <widget class="QLabel" name="vertexLabel">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>10</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Vertex</string>
   </property>
  </widget>
  <widget class="QPushButton" name="splitEdgeButton">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>400</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <italic>false</italic>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Split Edge</string>
   </property>
  </widget>
  <widget class="QPushButton" name="triangulateFaceButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>400</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Split Face</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>40</y>
     <width>421</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Corbel</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Please use LMB to select vertex, RMB to select half edge and MMB to select face.</string>
   </property>
  </widget>
  <widget class="QPushButton" name="subdivideButton">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>400</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Subdivide</string>
   </property>
  </widget>
  <widget class="QPushButton" name="extrudeButton">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>400</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Extrude Face</string>
   </property>
  </widget>
  <widget class="QPushButton" name="bevelEdgeButton">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>400</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>Bevel Edge</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="rDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>440</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="singleStep">
    <double>0.010000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="gDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="singleStep">
    <double>0.010000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="rLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>R</string>
   </property>
  </widget>
  <widget class="QLabel" name="gLabel">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>G</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="bDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="singleStep">
    <double>0.010000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="bLabel">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>B</string>
   </property>
  </widget>
  <widget class="QLabel" name="xLabel">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>X</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="xDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-100000000000000000620008645040778319495168.000000000000000</double>
   </property>
   <property name="maximum">
    <double>10000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815104.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="yLabel">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Y</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="yDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-100000000000000000620008645040778319495168.000000000000000</double>
   </property>
   <property name="maximum">
    <double>10000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815104.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="zLabel">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>440</y>
     <width>16</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Z</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="zDoubleSpinBox">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>440</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-100000000000000000620008645040778319495168.000000000000000</double>
   </property>
   <property name="maximum">
    <double>10000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815104.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="pushPullPPButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>470</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ebrima</family>
     <pointsize>10</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>PushPull++</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPDX">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>470</y>
     <width>62</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-100000000000000004764729344.000000000000000</double>
   </property>
   <property name="maximum">
    <double>10000000000000000303786028427003666890752.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPDY">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>470</y>
     <width>62</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPDZ">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>470</y>
     <width>62</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPTheta">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>470</y>
     <width>62</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPX">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPY">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPZ">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPNX">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPNY">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="pushPullPPPNZ">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>510</y>
     <width>62</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="minimum">
    <double>-1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1000000000000000015559416129466843024268201396921061433369770580430833781164755703264985389915047447676206280867840.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="dxLabel">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>470</y>
     <width>21</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>DX</string>
   </property>
  </widget>
  <widget class="QLabel" name="dyLabel">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>470</y>
     <width>21</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>DY</string>
   </property>
  </widget>
  <widget class="QLabel" name="dzLabel">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>470</y>
     <width>21</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>DZ</string>
   </property>
  </widget>
  <widget class="QLabel" name="thetaLabel">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>470</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Theta</string>
   </property>
  </widget>
  <widget class="QLabel" name="planeLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>510</y>
     <width>41</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Plane</string>
   </property>
  </widget>
  <widget class="QLabel" name="planeNormalLabel">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>510</y>
     <width>51</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Fixedsys</family>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Normal</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
